huaweicloud/services/waf/resource_huaweicloud_waf_cloud_instance.go

```diff
@@ -181,6 +176,84 @@ func resourceCloudInstanceCreate(ctx context.Context, d *schema.ResourceData, me
 		return diag.Errorf("error creating WAF v1 client: %s", err)
 	}
 
+	if d.Get("charging_mode").(string) == "prePaid" {
+		if err := common.ValidatePrePaidChargeInfo(d); err != nil {
+			return diag.FromErr(err)
+		}
+		if err := ValidatePrePaidParameter(d); err != nil {
+			return diag.FromErr(err)
+		}
+
+		orderId, err := createPrePaidCloudInstance(wafClient, cfg, region, d)
+		if err != nil {
+			return diag.Errorf("error creating pre paid cloud WAF: %s", err)
+		}
+		if orderId == nil {
+			return diag.Errorf("error creating pre paid cloud WAF, cause cannot find order id in response")
+		}
+
+		bssClient, err := cfg.BssV2Client(region)
+		if err != nil {
+			return diag.Errorf("error creating BSS v2 client: %s", err)
+		}
+		err = common.WaitOrderComplete(ctx, bssClient, *orderId, d.Timeout(schema.TimeoutCreate))
+		if err != nil {
+			return diag.Errorf("the order is not completed while creating pre paid cloud WAF: %s", err)
+		}
+		resourceId, err := common.WaitOrderResourceComplete(ctx, bssClient, *orderId, d.Timeout(schema.TimeoutCreate))
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		d.SetId(resourceId)
+	} else {
+		if err := ValidatePostPaidParameter(d); err != nil {
+			return diag.FromErr(err)
+		}
+
+		instance, err := createPostPaidCloudInstance(wafClient, cfg, region, d)
+		if err != nil {
+			return diag.Errorf("error creating post paid cloud WAF: %s", err)
+		}
+		if instance == nil {
+			return diag.Errorf("error creating post paid cloud WAF, cause cannot find instance in response")
+		}
+
+		resourceId, err := flattenResourceIdByType(instance, ResourceTypePayPerUseDomain)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		d.SetId(resourceId)
+	}
+
+	return resourceCloudInstanceRead(ctx, d, meta)
+}
+
+func ValidatePrePaidParameter(d *schema.ResourceData) error {
+	if _, ok := d.GetOk("website"); ok {
+		return fmt.Errorf("`website` cannot be specified in pre paid charging mode")
+	}
+	return nil
+}
+
+func ValidatePostPaidParameter(d *schema.ResourceData) error {
```

Comment:
These functions does not used outside of this package, please change these function protection level to the `private`.

Reply:
Done